server/src/main/java/io/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -1118,16 +1094,18 @@ public String apply(SegmentIdentifier input)
       final boolean removeOnDiskData
   )
   {
+    if (sink.isWritable()) {
```

Comment:
For one, it seemed to be more legit to check that to match the `we only count active sinks` comment which implies writable to me. Additionally, since there are a couple of paths to decrementing the `totalRows` counter, I wanted to make sure I never double decremented it (it's other decrement is also tied to finishWriting).

Reply:
Hmm, to prevent races (abandonSegment _could_ be called in a separate thread) how about having `sink.finishWriting()` return true or false, corresponding to its writability state before the method was called? Then, only decrement the counters if it returns true.

Reply:
:+1: